source/dotnet/Services/Infrastructure/Processes/ProcessStepResultRepository.cs

```diff
@@ -83,9 +83,21 @@ private async Task<ProcessStepResult> GetResultAsync(Guid batchId, string direct
                 point => new TimeSeriesPoint(
                     DateTimeOffset.Parse(point.quarter_time),
                     decimal.Parse(point.quantity, CultureInfo.InvariantCulture),
-                    point.quality))
+                    MapQuality(point.quality)))
             .ToList();
 
         return new ProcessStepResult(timeSeriesType, pointsDto.ToArray());
     }
+
+    private static TimeSeriesPointQuality MapQuality(string pointQuality)
+    {
+        return pointQuality switch
+        {
+            "measured" => TimeSeriesPointQuality.Measured,
+            "calculated" => TimeSeriesPointQuality.Calculated,
+            "estimated" => TimeSeriesPointQuality.Estimated,
+            "Incomplete" => TimeSeriesPointQuality.Incomplete,
+            _ => TimeSeriesPointQuality.Missing,
```

Comment:
Sure about this? How about
```suggestion
            "missing" => TimeSeriesPointQuality.Missing,
            _ => throw ...,
```